File: market/common-test/src/main/java/ru/yandex/market/wms/common/spring/TestSecurityDataProvider.java
package ru.yandex.market.wms.common.spring;

import java.util.Set;

import lombok.Data;
import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Component;

import ru.yandex.market.wms.common.model.enums.InforRole;
import ru.yandex.market.wms.shared.libs.authorization.SecurityDataProvider;
import ru.yandex.market.wms.shared.libs.env.conifg.Profiles;

@Data
@Profile(Profiles.TEST)
@Component
public class TestSecurityDataProvider implements SecurityDataProvider {
    private String user = "TEST";
    private String token = "TEST_TOKEN";
    private Set<String> roles = Set.of(InforRole.PACKING, InforRole.PACKING_WITHOUT_SCAN);
}

File: market/test/kotlin/ru/yandex/market/wms/inventory_management/config/TestSecurityDataProvider.kt
package ru.yandex.market.wms.inventory_management.config

import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Component
import ru.yandex.market.wms.shared.libs.env.conifg.Profiles

@Profile(ru.yandex.market.wms.shared.libs.env.conifg.Profiles.TEST)
@Component
class TestSecurityDataProvider : ru.yandex.market.wms.shared.libs.authorization.SecurityDataProvider {
    override fun getUser(): String {
        return "TEST"
    }

    override fun getToken(): String {
        return "TEST_TOKEN"
    }

    override fun getRoles(): MutableSet<String> {
        return HashSet()
    }
}
